lib/api/dsl/methods.js

```diff
@@ -856,11 +858,12 @@ function buildCurriedFunction(index, collection, field, operatorName, value, cur
   }
 
   if (!methods.dsl.filtersTree[index][collection].fields[field][curriedFunctionName]) {
-    curriedFunction = _.curry(operators[operatorName]);
-    curriedFunction = _.curry(curriedFunction(field, value));
-    if (not) {
-      curriedFunction = _.negate(curriedFunction);
-    }
+    // we need the ugly eval code to make sure the function can later on be serialized and
```

Comment:
I disagree with the word "ugly" (but I'm fine with your comment): `eval` should be used with extreme care and only if no other option is possible, as is the case here, so... I see no problem (it reminds me of old debates about the `goto` instruction :stuck_out_tongue: )